examples/shopping_cart.rs

```diff
@@ -0,0 +1,72 @@
+#![feature(type_alias_enum_variants)]
+
+#[macro_use]
+extern crate mirai_annotations;
+
+pub mod foreign_contracts;
+
+struct Cart {
+    items: Vec<i32>
+}
+
+impl Cart {
+    fn new() -> Cart {
+        Cart {
+            items: Vec::new(),
+        }
+    }
+
+    fn add_item(&mut self, value: i32) {
+        precondition!(value > 0);
+        let old_len = self.items.len();
+        self.items.push(value);
+        verify!(self.items.len() == old_len + 1);
+    }
+
+    fn remove_item(&mut self) -> Option<i32> {
+        let old_len = self.items.len();
+        let res = self.items.pop();
+        verify!(self.items.len() == 0 || self.items.len() == old_len - 1);
+        res
+    }
+
+    fn num_items(&self) -> usize {
+        self.items.len()
+    }
+
+    fn checkout(&self) -> i32 {
+        let mut sum: i32 = 0;
+        let len = self.items.len();
+        let mut i = 0;
+        while i < len {
+            let x = self.items.get(i);
+            match x {
+                None => {},
+                Some(item) => {
+                    sum = sum + *item;
+                }
+            }
+            i += 1;
+        }
+        verify!(len <= 0 || sum > 0);
```

Comment:
This is appropriate. It succinctly states a post condition for checkout. Making it explicit protects agains inadvertent relaxation of the post condition later on.